### PR TITLE
ovl_En_Bee OK

### PR DESCRIPTION
**Pull request by sonicdcer**

### CODE REVIEW

**Comment by krm01** [comment]
Draft of doc pass for this actor, will PR after this one gets merged: https://github.com/sonicdcer/mm/compare/ovl_En_Bee...krm01:mm:doc-giant-bee

File: src/overlays/actors/ovl_En_Bee/z_en_bee.c [hunk 1]
```diff
@@ -67,34 +69,219 @@ static DamageTable D_80B5B214 = {
...
+        OCELEM_ON,
+    },
     { 6, 13, -4, { 0, 0, 0 } },
 };
 
-#endif
+void EnBee_Init(Actor* thisx, PlayState* play) {
+    EnBee* this = THIS;
+
+    this->actor.colChkInfo.mass = 0xA;
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Bee/z_en_bee.c:95]: We use dec for mass iirc
```suggestion
    this->actor.colChkInfo.mass = 10;
```

File: src/overlays/actors/ovl_En_Bee/z_en_bee.c [hunk 2]
```diff
@@ -67,34 +69,219 @@ static DamageTable D_80B5B214 = {
...
+    this->unk_204 = Rand_S16Offset(0x14, 0x1E);
+    this->unk_208 = 0;
+    this->actionFunc = func_80B5A9E8;
+}
+
+void func_80B5A9E8(EnBee* this, PlayState* play) {
+    Vec3f sp3C;
+    s32 pad[2];
+
+    if ((this->actor.category != 5) && (ActorCutscene_GetCurrentIndex() == -1)) {
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Bee/z_en_bee.c:153]:
```suggestion
    if ((this->actor.category != ACTORCAT_ENEMY) && (ActorCutscene_GetCurrentIndex() == -1)) {
```

File: src/overlays/actors/ovl_En_Bee/z_en_bee.c [hunk 3]
```diff
@@ -67,34 +69,219 @@ static DamageTable D_80B5B214 = {
...
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bee/EnBee_Init.s")
+void EnBee_Update(Actor* thisx, PlayState* play) {
+    s32 pad;
+    EnBee* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bee/EnBee_Destroy.s")
+    SkelAnime_Update(&this->skelAnime);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bee/func_80B5A854.s")
+    if (this->actor.category == 5) {
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Bee/z_en_bee.c:255]:
```suggestion
    if (this->actor.category == ACTORCAT_ENEMY) {
```

File: src/overlays/actors/ovl_En_Bee/z_en_bee.h [hunk 4]
```diff
@@ -2,16 +2,28 @@
...
-    /* 0x204 */ char unk_204[0x84];
+    /* 0x204 */ s16 unk_204;
+    /* 0x206 */ s16 unk_206;
+    /* 0x208 */ s16 unk_208;
+    /* 0x20C */ s32 unk_20C;
+    /* 0x210 */ s32 unk_210; 
+    /* 0x214 */ s32 unk_214;
+    /* 0x218 */ s32 unk_218;
+    /* 0x21C */ Vec3f unk_21C[2];
+    /* 0x234 */ char pad234[8];
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Bee/z_en_bee.h:25]:
```suggestion
    /* 0x234 */ UNK_TYPE1 pad234[8];
```

**Review by Kenix3 — APPROVED** [review]

File: src/overlays/actors/ovl_En_Bee/z_en_bee.c [hunk 5]
```diff
@@ -67,34 +69,219 @@ static DamageTable D_80B5B214 = {
...
+
+    Math_Vec3f_Copy(&sp88, &player->actor.world.pos);
+    phi_fs1 = (this->unk_218 * 0x700) + 0x2000;
+
+    for (i = 0; i < 2; i++) {
+        rnd = randPlusMinusPoint5Scaled(20.0f);
+        sp88.x += Math_SinS((this->actor.yawTowardsPlayer + ((f32)this->unk_20C)) + phi_fs1) * (rnd + 30.0f);
+        sp88.y = (Math_SinS(this->unk_210) * 10.0f) + (player->actor.floorHeight + 40.0f);
+        rnd = randPlusMinusPoint5Scaled(20.0f);
+        sp88.z += Math_CosS((((f32)this->actor.yawTowardsPlayer) + this->unk_20C) + phi_fs1) * (rnd + 30.0f);
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Bee/z_en_bee.c:206]:
```suggestion
        sp88.z += Math_CosS(((f32)this->actor.yawTowardsPlayer + this->unk_20C) + phi_fs1) * (rnd + 30.0f);
```

**Review by hensldm — APPROVED** [review]
Just 1 small nit. Nice work.

**Pull request merged**